### PR TITLE
Feature/1 move license checker to repo

### LINKED ISSUE #1: Move existing code to this repo
Currently the code resides in https://github.com/cloudogu/confluence and should move here for a better building experience.

See https://github.com/cloudogu/confluence/issues/32

### PR DESCRIPTION
**Pull request by ppxl**
This PR moves the license-checker code from the confuence repo to this repo and makes some necessary adjustments

resolves #1

### CODE REVIEW

File: license/watcher/watcher.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	cmdExecutor   executor
+	licenseTester tester.Tester
+}
+
+// Watch watches in a fixed interval for license changes.
+func (dw *defaultWatcher) Watch() error {
+	duration := time.Duration(dw.args.WatchIntervalInSecs) * time.Second
+	log.Debugf("Start License check using %d seconds", dw.args.WatchIntervalInSecs)
+
+	for range time.Tick(duration) {
```
Comment (cesmarvin) [inline license/watcher/watcher.go:62]: **[checkstyle]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub>
SA1015: using time.Tick leaks the underlying ticker, consider using it only in endless functions, tests and the main package, and use time.NewTicker here

Comment (ppxl) [reply license/watcher/watcher.go:62]: resolved by ignoring. Given its purposes (trivial) and its runtime (short) using `time.Ticker` is okay compared to its more complex alternative:

```
time.Tick(duration)
...
vs
...
ticker := time.NewTicker(time.Second)
defer ticker.Stop() 
done := make(chan bool) 
go func() {
    time.Sleep(10 * time.Second)
    done <- true 
}() 
for {
    select {
    case <-done:
        fmt.Println("Done!")
        return
    case t := <-ticker.C:
        fmt.Println("Current time: ", t)
    }
}
```

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+String repositoryOwner
+String repositoryName
+String project
+String goProject
+String githubCredentialsId
+
+void make(String goal) {
+    sh "cd /go/src/${project} && make ${goal}"
+}
```
Comment (sklein94) [inline Jenkinsfile]: Github complaints about a missing newline at the end of the file

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            withSonarQubeEnv {
+                sh "${scannerHome}/bin/sonar-scanner -Dsonar.projectName=${repositoryName}:${env.BRANCH_NAME} -Dsonar.projectKey=$repositoryName -Dsonar.sources=. -Dsonar.junit.reportsPath=target/unit-tests/report.xml -Dsonar.go.coverage.reportPaths=target/unit-tests/coverage.out -Dsonar.exclusions=**/cmd/**,**/integrationTests/**,**/resources/**,**/*_test.go,**/vendor/**,report.xml -Dsonar.tests=. -Dsonar.test.inclusions=**/*_test.go -Dsonar.test.exclusions=**/vendor/**"
+            }
+            timeout(time: 2, unit: 'MINUTES') { // Needed when there is no webhook for example
+                def qGate = waitForQualityGate()
+                if (qGate.status != 'OK') {
+                    unstable("Pipeline unstable due to SonarQube quality gate failure")
+                }
+            }
+        }
```
Comment (sklein94) [inline Jenkinsfile]: I think we should use the sonar branch plugin here. You can almost copy it from the jenkinsfile in cesapp or doguctl.

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                    withEnv(["CI_PULL_REQUEST=${env.CHANGE_ID}", "CI_COMMIT=${commitSha}", "CI_REPO_OWNER=cloudogu", "CI_REPO_NAME=${repositoryName}"]) {
+                        make 'static-analysis'
+                    }
+                }
+            }
+        }
+        stage('SonarQube') {
+            def scannerHome = tool name: 'sonar-scanner', type: 'hudson.plugins.sonar.SonarRunnerInstallation'
+            withSonarQubeEnv {
+                sh "${scannerHome}/bin/sonar-scanner -Dsonar.projectName=${repositoryName}:${env.BRANCH_NAME} -Dsonar.projectKey=$repositoryName -Dsonar.sources=. -Dsonar.junit.reportsPath=target/unit-tests/report.xml -Dsonar.go.coverage.reportPaths=target/unit-tests/coverage.out -Dsonar.exclusions=**/cmd/**,**/integrationTests/**,**/resources/**,**/*_test.go,**/vendor/**,report.xml -Dsonar.tests=. -Dsonar.test.inclusions=**/*_test.go -Dsonar.test.exclusions=**/vendor/**"
```
Comment (sklein94) [inline Jenkinsfile]: To make this very long line shorter you can use a `sonar-project.properties`-file for `sonar.exclusions`, `sonar.test.exclusions`, `sonar.test.inclusions`, `junit.reportPaths`, `go.coverage.reportPaths`

**Review by sklein94 — CHANGES_REQUESTED** [review]

**Pull request merged**